### DISCUSSION TITLE
How to create custom singleton service and inject to other services?

### DISCUSSION THREAD

**Discussion by eric-makro** [medusajs/medusa#2406] [Q&A] [answered]
How to create custom singleton service and inject to other services?

**Comment by srindom** [comment, marked as answer]
We have creation of custom services documented here: https://docs.medusajs.com/advanced/backend/services/create-service

In your Medusa project you can place a file at `/src/services/cool-stuff.js` that default exports a service.

This service will automatically be registered in Medusa's dependency container as `coolStuffService`. 

To use it in other services you would do something like:
```js
class SomeService {
  // ...
  constructor({ coolStuffService }) {
    // ...
  }
}
```

To use it in an endpoint controller you can do:
```js
const coolStuffService = req.scope.resolve("coolStuffService`)
```

**Reply by eric-makro** [reply]
Thanks @srindom
